js/js.libraries/src/core/date.kt

```diff
@@ -4,5 +4,110 @@ package kotlin.js
+    public fun setMilliseconds(millis: Int): Long = definedExternally
+    public fun setMinutes(minutes: Int,
+                          seconds: Int = definedExternally,
+                          millis: Int = definedExternally): Long = definedExternally
+    public fun setMonth(month: Int,
+                        day: Int = definedExternally)
+    public fun setSeconds(seconds: Int,
+                          millis: Int = definedExternally): Long = definedExternally
+    public fun setTime(timeMillis: Long): Long = definedExternally
+    public fun setUTCDate(day: Int): Long = definedExternally
+    public fun setUTCFullYear(year: Int,
+                              month: Int,
+                              day: Int): Long = definedExternally
+    public fun setUTCHours(hours: Int,
+                           minutes: Int = definedExternally,
+                           seconds: Int = definedExternally,
+                           millis: Int): Long = definedExternally
+    public fun setUTCMilliseconds(millis: Int): Long = definedExternally
+    public fun setUTCMinutes(minutes: Int,
+                             seconds: Int = definedExternally,
+                             millis: Int = definedExternally): Long = definedExternally
+    public fun setUTCMonth(month: Int,
+                           day: Int = definedExternally): Long = definedExternally
+    public fun setUTCSeconds(seconds: Int,
+                             millis: Int = definedExternally): Long = definedExternally
+
+    @Deprecated("Be aware that this feature may cease to work at any time.", ReplaceWith("setFullYear"))
+    public fun setYear(year: Int): Long = definedExternally
+
+
+
+    public fun toDateString(): String = definedExternally
+    @Deprecated("Be aware that this feature may cease to work at any time.", ReplaceWith("toUTCString"))
+    public fun toGMTString(): String = definedExternally
+    public fun toISOString(): String = definedExternally
+    public fun toJSON(): String = definedExternally
+    public fun toLocaleDateString(locales: dynamic = definedExternally, /* String | Array<String> */
+                                  options: dynamic = definedExternally): String = definedExternally
+    public fun toLocaleString(locales: dynamic = definedExternally, /* String | Array<String> */
+                              options: dynamic = definedExternally): String = definedExternally
+    public fun toLocaleTimeString(locales: dynamic = definedExternally, /* String | Array<String> */
+                                  options: dynamic = definedExternally): String = definedExternally
+    public override fun toString(): String = definedExternally
+    public fun toTimeString(): String = definedExternally
+    public fun toUTCString(): String = definedExternally
+    public fun valueOf(): Long = definedExternally
+
+    companion object {
+        fun now(): Long = definedExternally
```

Comment:
Returns cannot be Int because, according to docs, all returns is a timestamp since 1970 in milliseconds. Also, it's bad to treat it is as Double, because you can't than convert it to Date without ugly cast .toLong(). Same thing with Number, ugly casts all over the code